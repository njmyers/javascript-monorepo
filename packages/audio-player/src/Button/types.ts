import * as React from 'react';
import { Props as IconProps } from '@fortawesome/react-fontawesome';

export interface Props extends IconProps {
  onClick: (e: React.MouseEvent<HTMLElement>, ...args: [any]) => void;
}
