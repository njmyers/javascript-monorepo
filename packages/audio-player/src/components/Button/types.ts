import { Props as IconProps } from '@fortawesome/react-fontawesome';

export type ButtonProps = JSX.IntrinsicElements['button'];
export interface Props extends ButtonProps {
  icon: IconProps['icon'];
  size: IconProps['size'];
}
