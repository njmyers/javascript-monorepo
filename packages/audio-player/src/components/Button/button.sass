@import ../../global

.audioPlayer
  button
    border-radius: 100%
    padding: 0
    height: 45px
    width: 45px
    margin: 0 2px
    box-sizing: border-box
    border: none
    cursor: pointer
    color: $white
    background-color: $secondary-color

    +transition(0.2s opacity)

    &:focus
      outline: none

    &:hover
      opacity: 0.8

    &.backward
      padding-right: 3px

    &.forward
      padding-left: 4px

    &.play
      width: 55px
      height: 55px
      padding-left: 4px

    &.pause
      width: 55px
      height: 55px
      padding-left: 0px

    &.stop
      width: 55px
      height: 55px
