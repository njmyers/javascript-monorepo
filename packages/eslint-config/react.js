module.exports = {
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    'shared-node-browser': true,
  },
  extends: ['plugin:react/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  rules: {
    'jsx-a11y/label-has-associated-control': 'error',
    'react/display-name': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
