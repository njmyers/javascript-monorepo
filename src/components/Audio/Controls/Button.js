import React from 'react';

const Button = ({ icon = 'play', size = 'lg', className = 'audio-button', cb } = {}) => {
	const faClass = `fa fa-${icon} fa-${size}`;
	const classes = `${className} ${faClass}`;

	return <button className={classes} onClick={cb} />;
};

export default Button;
