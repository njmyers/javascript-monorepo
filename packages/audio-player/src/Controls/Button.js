import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import './button.sass';

const Button = ({ icon = 'play', className, onClick, size } = {}) => {
  return (
    <button className={`audioPlayer_button ${className}`} onClick={onClick}>
      <FontAwesomeIcon icon={icon} size={size} />
    </button>
  );
};

export default Button;
