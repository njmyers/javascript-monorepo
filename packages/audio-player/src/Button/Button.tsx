// @ts-nocheck
import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Props } from './types';

import './button.sass';

const Button = ({ onClick, icon, ...props }: Props) => {
  const name = icon;

  return (
    <button
      type="button"
      className={name}
      name={name}
      aria-label={name}
      onClick={onClick}
    >
      <FontAwesomeIcon icon={icon} {...props} />
    </button>
  );
};

export default Button;
