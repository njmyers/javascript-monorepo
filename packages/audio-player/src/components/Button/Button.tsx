import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Props } from './types';

import './button.sass';

const Button: React.FC<Props> = ({ onClick, icon, ...props }) => {
  const name = icon;

  return (
    <button
      type="button"
      className={name}
      name={name}
      aria-label={name}
      onClick={onClick}
    >
      <FontAwesomeIcon icon={icon} {...props} />
    </button>
  );
};

export default Button;
