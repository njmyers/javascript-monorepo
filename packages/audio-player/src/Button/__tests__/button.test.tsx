import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from '../Button';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Button />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
