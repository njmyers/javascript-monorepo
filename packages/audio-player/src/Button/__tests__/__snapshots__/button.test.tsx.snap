// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<button
  type="button"
/>
`;
